tests/settings_tests/tests.py

```diff
@@ -441,3 +441,130 @@ def test_tuple_settings(self):
+        It fails while certain settings are being changed.
+
+        - SETTING_BOTH raises an error while receiving the signal
+          on both entering and exiting the context manager
+        - SETTING_ENTER raises an error only on enter.
+        - SETTING_EXIT raises an error only on exit.
+        """
+        setting = kwargs['setting']
+        enter = kwargs['enter']
+
+        if setting in ('SETTING_BOTH', 'SETTING_ENTER') and enter:
+            raise SettingChangeEnterException
+        if setting in ('SETTING_BOTH', 'SETTING_EXIT') and not enter:
+            raise SettingChangeExitException
+
+    def check_settings(self):
+        """
+        Ensure that settings for this test case are missing from the settings module.
+        """
+        self.assertFalse(hasattr(settings, 'SETTING_BOTH'))
+        self.assertFalse(hasattr(settings, 'SETTING_ENTER'))
+        self.assertFalse(hasattr(settings, 'SETTING_EXIT'))
+        self.assertFalse(hasattr(settings, 'SETTING_PASS'))
+
+    def check_spy_receiver_exit_calls(self, call_count):
+        """
+        Ensure that `self.spy_receiver` was called exactly `call_count` times
+        with ``enter=False`` keyword argument.
+        """
+        kwargs_with_exit = [kwargs for args, kwargs in self.spy_receiver.call_args_list
+                            if ('enter', False) in kwargs.items()]
+
+        self.assertEqual(len(kwargs_with_exit), call_count)
+
+    def test_override_settings_both(self):
+        """
+        Receiver fails on both enter and exit.
+        """
+        with self.assertRaises(SettingChangeEnterException):
+            with override_settings(SETTING_PASS='BOTH', SETTING_BOTH='BOTH'):
+                pass
+
+        self.check_settings()
+
+        # Two settings were touched, so expect two calls of `spy_receiver`.
+        self.check_spy_receiver_exit_calls(call_count=2)
+
+    def test_override_settings_enter(self):
+        """
+        Receiver fails on enter only.
+        """
+        with self.assertRaises(SettingChangeEnterException):
+            with override_settings(SETTING_PASS='ENTER', SETTING_ENTER='ENTER'):
+                pass
+
+        self.check_settings()
+
+        # Two settings were touched, so expect two calls of `spy_receiver`.
+        self.check_spy_receiver_exit_calls(call_count=2)
+
+    def test_override_settings_exit(self):
+        """
+        Receiver fails on exit only.
+        """
+        with self.assertRaises(SettingChangeExitException):
+            with override_settings(SETTING_PASS='EXIT', SETTING_EXIT='EXIT'):
+                pass
+
+        self.check_settings()
+
+        # Two settings were touched, so expect two calls of `spy_receiver`.
+        self.check_spy_receiver_exit_calls(call_count=2)
+
+    def test_override_settings_reusable_on_enter(self):
+        """
+        Error should be raised correctly
+        when reusing the same override_settings instance.
```

Comment:
Can you re-warp this block to 79 chars? (First line is too short.)

Reply:
Sure, I will. I preferred it this way, because I'm a big believer of [semantic line break](http://sembr.org/).